rs/moq-relay/src/reload.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::{Arc, Mutex};
+#[cfg(unix)]
+use tokio::signal::unix::{signal, SignalKind};
+
+type ReloadListener = dyn Fn() + Send + Sync;
+
+pub struct ConfigReloader {
```

Comment:
I don't think you need this struct at all. It makes things more complicated IMO.

I woild replace it with:

```rs
#[cfg(unix)]
tokio::spawn(async move {
    let mut sigusr1 = tokio::signal::unix::signal(SignalKind::user_defined1())?;
    while let Some(_) = sigusr1.recv() {
       // contents of listener
    }
});
```

Even if we had multiple reloadable configurations, you could spawn a task for each. Tokio fans out signals.